From: Openkylin Developers <packaging@lists.openkylin.top>
Date: Thu, 8 Sep 2022 02:10:22 +0000
Subject: Turn #error into exceptions

So the library can be built on non-Linux platforms too, although can't
guarauntee its functionality regarding that piece of code.
---
 base/file.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/base/file.cpp b/base/file.cpp
index 3af7322..e25bd19 100644
--- a/base/file.cpp
+++ b/base/file.cpp
@@ -253,7 +253,8 @@ std::string GetExecutablePath() {
   path[PATH_MAX - 1] = 0;
   return path;
 #else
-#error unknown OS
+#include <stdexcept>
+  throw std::runtime_error(std::string("Unknown OS!"));
 #endif
 }
 
